text/0000-separate-context-propagation.md

```diff
@@ -0,0 +1,195 @@
+
+**CleanBaggage( context) context**
+To avoid sending baggage to an untrusted downstream process, the Baggage API provides a function remove all baggage from a context, 
+
+**GetPropagator( type) inject, extract**
+To register with the propagation system, the Baggage API provides a set of propagation functions for every propagation type.
+
+
+## Additional APIs
+
+Because the application and context propagation layers are separated, it is possible to create new distributed applications which do not depend on either the Observability or Baggage APIs.
+
+**GetPropagator(type) inject, extract**
+To register with the propagation system, additional APIs provide a set of propagation functions for every propagation type.
+
+
+# Context Propagation Layer
+
+## Context API
+
+Distributed applications access data in-process using a shared context object. Each distributed application sets a single key in the context, containing all of the data for that system.
+
+**SetValue( context, key, value) context**  
+To record the local state of an application, the Context API provides a function which takes a context, a key, and a value as input, and returns an updated context which contains the new value.
+
+**GetValue( context, key) value**  
+To access the local state of an application, the Context API provides a function which takes a context and a key as input, and returns a value.
+
+**Optional: Automated Context Management**  
+When possible, context should automatically be associated with program execution . Note that some languages do not provide any facility for setting and getting a current context. In these cases, the user is responsible for managing the current context. 
+
+**Optional: SetCurrentContext( context)**  
+To associate a context with program execution, the Context API provides a function which takes a Context.
+
+**Optional: GetCurrentContext() context**  
+To access the context associated with program execution, the Context API provides a function which takes no arguments and returns a Context.
+
+
+## Propagation API
+
+Distributed applications send data to downstream processes via propagators, functions which read and write application context into RPC requests. Each distributed application creates a set of propagators for every type of supported medium - currently HTTP and Binary.
+
+**Inject( context, request)**  
+To send the data for all distributed applications downstream to the next process, the Propagation API provides a function which takes a context and a request, and mutates the request to include the encoded context. The canonical representation of a request is as a map.
+
+**Extract( context, request) context**  
+To receive the data from all distributed applications set by prior upstream processes, the Propagation API provides a function which takes a context and a request, and mutates the request to include the encoded context. The canonical representation of a request is as a map.
```

Comment:
Everything after the first comma is identical to the corresponding `Inject` text. I think you mean it `mutates the context`?

Reply:
Fixed. In general, the descriptions of the functions could be better, I'll improve them this week. Please let me know if the functions themselves capture everything.